From 3e2c6735eaab8795bd213cb90eef853a322e2924 Mon Sep 17 00:00:00 2001
From: Yuvaraj Ranganathan <quic_yrangana@quicinc.com>
Date: Wed, 27 Nov 2024 16:21:57 +0530
Subject: [PATCH] FROMLIST: arm64: dts: qcom: qcs8300: add TRNG node

The qcs8300 SoC has a True Random Number Generator, add the node with
the correct compatible set.

Reviewed-by: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
Signed-off-by: Yuvaraj Ranganathan <quic_yrangana@quicinc.com>
Upstream-Status: Submitted [https://lore.kernel.org/all/20241125064317.1748451-3-quic_yrangana@quicinc.com/]
---
 arch/arm64/boot/dts/qcom/qcs8300.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/qcs8300.dtsi b/arch/arm64/boot/dts/qcom/qcs8300.dtsi
index 0aca3550e8c0..dcdeb0144af9 100644
--- a/arch/arm64/boot/dts/qcom/qcs8300.dtsi
+++ b/arch/arm64/boot/dts/qcom/qcs8300.dtsi
@@ -610,6 +610,11 @@ &clk_virt SLAVE_QUP_CORE_0 0>,
 			};
 		};
 
+		rng: rng@10d2000 {
+			compatible = "qcom,qcs8300-trng", "qcom,trng";
+			reg = <0x0 0x010d2000 0x0 0x1000>;
+		};
+
 		config_noc: interconnect@14c0000 {
 			compatible = "qcom,qcs8300-config-noc";
 			reg = <0x0 0x014c0000 0x0 0x13080>;
-- 
2.34.1

